src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1565,3 +1583,18 @@ int sd_dhcp_server_set_relay_agent_information(
         free_and_replace(server->agent_remote_id, remote_id_dup);
         return 0;
 }
+
+int sd_dhcp_server_set_lease_file(sd_dhcp_server *server, const char *path) {
+        assert_return(server, -EINVAL);
+        assert_return(!sd_dhcp_server_is_running(server), -EBUSY);
+
+        if (!path) {
+                server->lease_file = mfree(server->lease_file);
+                return 0;
+        }
+
+        if (!path_is_safe(path))
+                return -EINVAL;
```

Comment:
Please change this to `if (path && !path_is_safe(path))` and drop the separate handling of `if (!path)`. This way, we get the expected return value 1 if the value changes, and 0 otherwise. (And the code is simpler.)